rootex/framework/systems/render_system.cpp

```diff
@@ -107,6 +107,21 @@ void RenderSystem::renderPassRender(float deltaMilliseconds, RenderPass renderPa
 			mc->postRender();
 		}
 	}
+
+	AnimatedModelComponent* amc = nullptr;
+	for (auto& component : s_Components[AnimatedModelComponent::s_ID])
+	{
+		amc = (AnimatedModelComponent*)component;
+		if (amc->getRenderPass() & (unsigned int)renderPass)
+		{
+			amc->preRender(deltaMilliseconds);
+			if (amc->isVisible())
+			{
+				amc->render();
+			}
+			amc->postRender();
+		}
+	}
```

Comment:
Now that all renderable components inherit `RenderableComponent`, why don't we loop over all the objects of `RenderableComponent` objects instead of individually going over the objects of the inherited classes?